### PR TITLE
Fix issue in Gemm Add fusion

### PR DESCRIPTION
**Pull request by philass**
Currently when `beta=0`. This will incorrectly fuse the add op.

`beta` should be set to 1 in the pattern that it is replaced with.

Co-authored-by: Zachary Cetinic <zcetinic@groq.com>
Signed-off-by: Philip Lassen <plassen@groq.com>

### CODE REVIEW

File: src/Dialect/ONNX/Rewrite.td [hunk 1]
```diff
@@ -254,7 +254,7 @@ def MulAddToGemmOptPattern : Pat<(ONNXAddOp (ONNXMatMulOp:$res $m1, $m2), $m3),
 
 // onnx.add(onnx.Gemm(%X, %Y, None), %Z) = onnx.Gemm(%X, %Y, %Z)
```
Comment (sorenlassen) [inline src/Dialect/ONNX/Rewrite.td:255]: I suggest that you make beta=1.0 clear here, e.g. by writing it as
```
// onnx.add(onnx.Gemm(%X, %Y, None), %Z) = onnx.Gemm(%X, %Y, %Z) {beta=1.0}
```

**Review by sorenlassen — APPROVED** [review]
LGTM

thanks for adding lit test

File: src/Dialect/ONNX/Rewrite.td (hunk 1 above)
Comment (philass) [reply src/Dialect/ONNX/Rewrite.td:255]: Good point! Thanks for the review

**Review by tungld — APPROVED** [review]
LGTM.

**Pull request merged**